### ISSUE TITLE
A few useful additions to wavfile.py (24 bit read/write support, cue markers support, loop makers support, etc.)

### ISSUE THREAD

**Issue by josephernest** [scipy/scipy#6849]
I've modified the file https://github.com/scipy/scipy/blob/master/scipy/io/wavfile.py along the past years to add:

Here is the result: https://gist.github.com/josephernest/3f22c5ed5dabf1815f16efa8fa53d476

If you're interested, feel free to reuse this / commit (I won't have time to do a pull request soon).

**Comment by rgommers** [comment]
Thanks @josephernest. 24-bit support would indeed be nice. cross-linking the issue about that: gh-1930.

**Comment by josephernest** [comment]
Thanks @rgommers 
I use my code (especially 24-bit part) in many of my projects, and I confirm that it works and has been widely tested.

____

More informations:

Normally `wavfile.py`works like this:

    (sr, samples) = read('hello.wav')

To keep backwards compatibility, I added parameters (default `False`) that state if we want extra metadatas while reading (`br`: bitrate, `cue`: list of markers, `f0`: frequency of the root note in Hz stored in some wav files (often the case in instrument sample sets), etc.):

    (sr, samples, br, cue, f0, cuechunkstr, smplchunkstr, cuelabels) = read('hello.wav', \
       readmarkers=True, readpitch=True, cuechunk=True, smplchunk=True, cuelabels=True)

Drawbacks of my code:

* not documented

* to go fast, I removed RIFX support (I have never seen a single RIFX big-endian WAV audio file in 10+ years of audio production / audio programming), you probably wouldn't like this

So in short, I'm open to do a pull request, but I'd need help :)

**Comment by perimosocordiae** [comment]
@josephernest I got this started in gh-6852. Could you contribute a small test file that exercises the additional functionality?

**Comment by josephernest** [comment]
Thanks @perimosocordiae , I'm currently very slow in the `git` workflow. By email or chat I could provide you the files in ten minutes, but by forking / commiting / pushing / doing pull requests etc. it would require me a few hours that I won't have the next 4 weeks.

____

[Here]() is a 24-bit 44.1 khz stereo WAV file named `test24bit.wav`. It has:

* a marker at 0'04", another marker at 0'08"
* an infinite loop from 0'12" to 0'16"
* a pitch metadata of A5 (Sony SoundForge notation) that is the standard A4, i.e. 440 Hz

With my [API proposal here](https://github.com/scipy/scipy/pull/6852#issuecomment-266532025),

    read('test24bit.wav', metadata=True)

should return:

    (44100, 
     array([[  ...  ]]), 
     {'loops': [[..., ...]], 'markers': [..., ....], 'pitch': 440.0, 'bitrate': 24})

**Comment by perimosocordiae** [comment]
Thanks, no git shenanigans required! The test file you provided is pretty large (5.1MB vs a couple KB for most of the existing test fixtures). Is it possible to make a similar one with a tiny amount of data to keep filesize low?

**Comment by josephernest** [comment]
(I wasn't a criticism about `git`, but just the current state of my skills in the `git` workflow)

[Here is a small test file](http://gget.it/f26wj6mq/test24bit_2.wav). I won't go below 250 milliseconds, because testing if a loop works wouldn't make sense with a 10 ms sound file:

**Comment by perimosocordiae** [comment]
Great, I've added this to the PR as a new test fixture. Thanks!

**Comment by josephernest** [comment]
Thanks!

**Comment by X-Raym** [comment]
The write function for loops was broken, and few other things with markers,

I made an updated version here https://github.com/X-Raym/wavfile.py

It also fix various issues, like the loss of metadata (unsupported chinks are now supported), have LIST - INFO etc...

**Comment by X-Raym** [comment]
@ilayn I will make further testing on my own, to be sure it works ok, and I'll do that, maybe in few days. Still, the script can be tested for those who needed (just download it from my repo).

My own version may not respect Scipy conventions, as I don't know a lot of thing about it, apart from the original wavfile.py file. And just like Joseph code, I didn't write any updated usage doc for now.

Thanks for your interest interest, I hope this will help make these features (especially 24bit support brought by @josephernest and unsupported chunks read/write I made) be merged in your project 🐱

**Comment by rudytrubitt** [comment]
@X-Raym ,  @ilayn , @josephernest , @perimosocordiae I'm wondering if anyone knows if this has moved forward since June of 2018. The ability to add markers would be particularly useful to me.

thanks,
rudy

**Comment by afparsons** [comment]
@endolith has a thrice-derived fork [here](https://github.com/endolith/wavfile.py/tree/metadata_develop) implementing `ltxt` extraction (a.k.a. "region" or "marker range"). I'd also love to see such support added.

**Comment by endolith** [comment]
(I have other WAV-related changes in my local repo, please bother me to make PRs out of them)
